internal/txapp/txapp.go

```diff
@@ -942,7 +942,7 @@ func (r *TxApp) checkAndSpend(ctx TxContext, tx *transactions.Transaction, price
 				return nil, transactions.CodeUnknownError, err2
 			}
 
-			return nil, transactions.CodeInsufficientBalance, fmt.Errorf("transaction tries to spend %s tokens, but account only has %s tokens", amt.String(), tx.Body.Fee.String())
+			return account.Balance, transactions.CodeInsufficientBalance, fmt.Errorf("transaction tries to spend %s tokens, but account only has %s tokens", amt.String(), tx.Body.Fee.String())
```

Comment:
Ah, for abci's `ExecTxResult`.  Yah, seems alright given `FinalizeBlock` is written with this flow in mind.

I'm assuming this altered result does not affect apphash in any way, just logging and meta data obtainable from cometbft via e.g. their RPC server?  Regardless, it's alright for 0.8.  Just curious.

Reply:
umm, yeah. should be okay for 0.8, but this would be a consensus change.

Reply:
It does affect app hash?  I had an idea that not everything does, but wasn't sure what.   e.g. Code obviously would, but "logs" and "events" etc. I wasn't sure about.  Let's assume this fix can only be on 0.8.  :)

Reply:
Oh wait, we had talked about this with @brennanjl before.

Although we have 

```go
		abciRes.Code = txRes.ResponseCode.Uint32()
		abciRes.GasUsed = txRes.Spend
```

we thought that we should keep cometbft blind to gas.  There were threads about this:

- https://github.com/cosmos/cosmos-sdk/discussions/18894
- ... (I know there was a cometbft one, maybe and adr, looking...)

What's your take on these @charithabandi ?

Reply:
Ah, this is old and in tendermint context: https://github.com/cometbft/cometbft/blob/main/docs/references/rfc/tendermint-core/rfc-011-delete-gas.md
Perhaps it's not relevant in cometbft anymore, but the more recent cosmos discussion suggests it's still the thinking. No?

Reply:
Ok, disregard all that.  Until and if cometbft gets rid of gas recording as part of the recorded transaction result, I think there's a strong case to keep it.  Namely, we don't have our own spend journal that can be used to report how much a transaction spent or the change of balance over time.

Reply:
Yeah, makes sense